src/linker/Linker.Steps/LinkAttributesParser.cs

```diff
@@ -91,10 +92,17 @@ static string FormatCustomAttribute (CustomAttribute ca)
 			}
 		}
 
-		TypeDefinition? GenerateRemoveAttributeInstancesAttribute ()
+		TypeDefinition? GenerateRemoveAttributeInstancesAttribute (int ctorArgumentCount)
 		{
-			if (_context.MarkedKnownMembers.RemoveAttributeInstancesAttributeDefinition != null)
-				return _context.MarkedKnownMembers.RemoveAttributeInstancesAttributeDefinition;
+			TypeDefinition? td = null;
+
+			if (_context.MarkedKnownMembers.RemoveAttributeInstancesAttributeDefinition is TypeDefinition knownTypeDef) {
+				if (knownTypeDef.GetConstructorsOnType (_ => true).Any (m => m.Parameters.Count == ctorArgumentCount)) {
```

Comment:
Why not apply the filter in the GetConstructorsOnType ?

Reply:
That's a good point, that would be simpler and probably allocate less.